Init

vincerubinetti opened the pull request.

vincerubinetti committed:
initial upload

vincerubinetti committed:
touchups

vincerubinetti committed:
nojekyll

vincerubinetti committed:
trigger netlify

vincerubinetti committed:
add viz, split up

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
-# software-registry
\ No newline at end of file
+# CU DBMI Software Registry
```
(aligning with other content)
```suggestion
# CU DBMI Wall of Software
```

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
+The [Department of Biomedical Informatics at the University of Colorado](https://medschool.cuanschutz.edu/dbmi) investigates complex problems in medicine and biology using integrated computational technology.
+This repository and website contains a non-comprehensive list of tools, packages, workflows, resources, and other software that have been developed by our diverse group researchers, scientists, developers, and collaborators in support of [our mission](https://medschool.cuanschutz.edu/dbmi/about-us).
+
+[⭐️⭐️ View the List ⭐️⭐️](https://CU-DBMI.github.io/software-registry)
```
```suggestion
[⭐️⭐️ View the List ⭐️⭐️](https://CU-DBMI.github.io/wall-of-software)
```

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
+## Submit Your Software
+
+If you work for or with the department, you are eligible to submit your software here.
+To start, [open a pull request in this repo](https://github.com/CU-DBMI/software-registry/pulls) with the following changes:
```
```suggestion
To start, [open a pull request in this repo](https://github.com/CU-DBMI/wall-of-software/pulls) with the following changes:
```

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
+
+## Submit Your Software
+
+If you work for or with the department, you are eligible to submit your software here.
```
Would it make sense to provide a note here that the Software Engineering Team could help with this if someone would like their work to appear here? Mostly I'm wondering this out of kindness for folks who may not understand the technology behind it (might be working in other capacities).

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
+  - A short ~10-30 char description.
+  - A link to a website or repo.
+- Add a _web-friendly_ image to `/images`.
+  - Filename **must** be lower-case, kebab-case version of entry name.
```
Consider adding in a link to describe kebab-case.
```suggestion
  - Filename **must** be lower-case, [kebab-case](https://developer.mozilla.org/en-US/docs/Glossary/Kebab_case) version of entry name.
```

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
+  - **Must** have extension `.jpg`.
+  - Roughly square.
+  - Enough margins for main content to fit nicely inside hexagon.
+  - Larger than ~400×400, smaller than ~1000×1000
```
```suggestion
  - Larger than ~400×400, smaller than ~1000×1000 pixels
```

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
+
+### Printed
+
+If you'd also like your image to be printed out and posted on the wall of the DBMI office (7th floor Anschutz Health Sciences Building), **also** include these changes:
```
```suggestion
If you'd also like your image to be printed out and posted on the wall of the DBMI office (6th floor Anschutz Health Sciences Building), **also** include these changes:
```

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
\ No newline at end of file
+# CU DBMI Software Registry
+
+The [Department of Biomedical Informatics at the University of Colorado](https://medschool.cuanschutz.edu/dbmi) investigates complex problems in medicine and biology using integrated computational technology.
```
```suggestion
The [Department of Biomedical Informatics (DBMI) at the University of Colorado](https://medschool.cuanschutz.edu/dbmi) investigates complex problems in medicine and biology using integrated computational technology.
```

d33bs commented on index.html:
```diff
@@ -0,0 +1,236 @@
+      property="og:description"
+      content="The many software produced by the CU Department of Biomedical Informatics."
+    />
+    <!-- <meta property="og:image" content="share.jpg" /> -->
```
Double checking, should this be removed?

vincerubinetti replied:
I still need to add this image.

d33bs commented on src/viz.js:
```diff
@@ -0,0 +1,292 @@
+import {
```
Consider adding (where relevant and syntactically correct) a description of what this file does or adds. For example `// Defines visual for banner of webpage`, or something similar. It may be helpful to do the same for `/src/alpine.js`.

vincerubinetti replied:
Before merging I was going to do one more pass to add more comments and such.

d33bs commented on index.html:
```diff
@@ -0,0 +1,236 @@
+                  </template>
+
+                  <template x-if="entry.description">
+                    <div x-text="entry.description" class="truncate"></div>
```
I wasn't sure where best to place this comment, apologies if this is not the correct location. When hovering over a software item, I noticed that the icon names and descriptions have their text limited in a way that might look confusing. I'm unsure if the text could be legible at a smaller size and still fit within the limitations of the space. Maybe just the title could be helpful / simple enough? Maybe this is just a personal problem for only me based on my browser settings 🙂 (if so, please feel free to ignore) .

![image](https://github.com/CU-DBMI/wall-of-software/assets/3738008/7d382d30-5a16-4e2b-a6d5-60403424c08f)

falquaddoomi replied:
I was about to report that, too. I think having just the title isn't a bad idea, perhaps with a tooltip or something to display the extra text.

vincerubinetti replied:
Will tweak this more. Is a balance between using the space efficiently and not overflowing the hexagon, because can't make it wrap exactly to the hexagon shape (without making it significantly more complicated).

d33bs commented on README.md:
```diff
@@ -1 +1,37 @@
-# software-registry
\ No newline at end of file
+# CU DBMI Software Registry
```
Naming things is a challenge! Openly wondering if another name would provide additional elegance or sense of pride in the work for the department. Mostly I'm imagining members of DBMI saying something like "Hey I just got added to the \<name goes here\> today!" (what would add excitement / enhancement to this?). To this point, I think moving from "Software Registry" to "Wall of Software" is a great change (getting "registered" isn't often exciting and might be a negative).

Other things that come to mind: "DBMI Software Tapestry", "Software Hall of Fame", "Software Achievement Wall".  I can also see how being more humble / frank about it would be better 🙂 (feel free to dismiss this as overly-semantic).

vincerubinetti replied:
Yeah I think WOS is more descriptive and more fun. Tapestry would also be cool. Wouldn't wan't to call it HOF just b/c if you're package isn't in there it may feel bad.

d33bs reviewed (commented):
Super cool work @vincerubinetti 🤩 ! I left a few comments here in the hopes that this helps things along. Please don't hesitate to let me know if you have any questions.

falquaddoomi reviewed (approved):
Nifty, looks like you had fun with this. I hadn't heard of Alpine before; that was educational. I don't think I have anything to say that Dave hasn't already, so I'll approve.

vincerubinetti committed:
incorp some review comments

vincerubinetti committed:
tune up

vincerubinetti committed:
more tweaks

vincerubinetti commented:
Okay this should be good to go. Moved the description from inside to a tooltip and fixed/tweaked some other things. @cgreene Lmk if there is some other feature or content you want to add here for now, otherwise we can do it later.

In the next PR I guess we'll remove the fake data and maybe manually populate a few entries ourselves.

vincerubinetti committed:
one last tweak

vincerubinetti committed:
one last tweak

cgreene commented:
WOW!
